conn.go

```diff
@@ -169,6 +169,15 @@ func (c *Conn) Query(ctx context.Context, name string) (dnsmessage.ResourceHeade
 	c.mu.Unlock()
 
 	defer ticker.Stop()
+	defer func() {
+		c.mu.Lock()
+		defer c.mu.Unlock()
+		for i := len(c.queries) - 1; i >= 0; i-- {
+			if c.queries[i].nameWithSuffix == nameWithSuffix {
+				c.queries = append(c.queries[:i], c.queries[i+1:]...)
+			}
+		}
```

Comment:
Similar logic used to cleanup when an answer is received.

Reply:
I think we may want to keep track of the query pointer value since this may invalidate someone else's query of the same name, right?

Reply:
Something like

```golang
	queryChan := make(chan queryResult, 1)
	c.mu.Lock()
	query := &query{nameWithSuffix, queryChan}
	c.queries = append(c.queries, query)
	c.mu.Unlock()

	defer func() {
		c.mu.Lock()
		defer c.mu.Unlock()
		for i := 0; i < len(c.queries); i++ {
			if c.queries[i] == query {
				c.queries = append(c.queries[:i], c.queries[i+1:]...)
			}
		}
	}()

	ticker := time.NewTicker(c.queryInterval)
	defer ticker.Stop()
```

Reply:
Good catch, I'll change it to work that way :+1: